libs/blocks/modal/modal.js

```diff
@@ -137,7 +152,12 @@ export async function getModal(details, custom) {
     [...document.querySelectorAll('header, main, footer')]
       .forEach((element) => element.setAttribute('aria-disabled', 'true'));
   }
-
+  if (dialog.classList.contains('commerce-frame')) {
+    const { debounce } = await import('../../utils/action.js');
+    window.addEventListener('message', (messageInfo) => {
+      debounce(sendViewportDimensionsOnRequest(messageInfo));
```

Comment:
You can remove the debounce from here, it does nothing. If you don't need this event listener after the initial setup, anymore consider cleaning it up and remove it - otherwise you continue to listen to uninteresting messages.
```suggestion
     sendViewportDimensionsOnRequest(messageInfo)
```

Reply:
if (dialog.classList.contains('commerce-frame')) {
    window.addEventListener('message', (messageInfo) => {
      if (messageInfo.data === 'viewportWidth') {
        sendViewportDimensionsOnRequest(messageInfo);
      }
    });
  }
 Moving the condition (messageInfo.data === 'viewportWidth') into the event listener would add the message event  listener every time a dialog modal is opened.